bonelab/cli/RapidPrototype.py

```diff
@@ -3,158 +3,978 @@
+  writer.Update()
+
+  message("Writing file " + output_file)
+
+def view_stl(input_file, transform_file, func):
+
+  model = vtk.vtkSTLReader()
+  model.SetFileName(input_file)
+  model.Update()
+  
+  model = applyTransform(transform_file, model)
+  
+  mat4x4 = visualize_actors( model.GetOutputPort(), None )
+  
+def add_stl(input_file1, input_file2, output_file, transform_file, visualize, overwrite, func):
+
+  if os.path.isfile(output_file) and not overwrite:
+    result = input('File \"{}\" already exists. Overwrite? [y/n]: '.format(output_file))
+    if result.lower() not in ['y', 'yes']:
+      print('Not overwriting. Exiting...')
+      os.sys.exit()
+  
+  im1 = vtk.vtkSTLReader()
+  im1.SetFileName(input_file1)
+  im1.Update()
+  
+  im2 = vtk.vtkSTLReader()
+  im2.SetFileName(input_file2)
+  im2.Update()
+
+  im2 = applyTransform(transform_file, im2)
+  
+  if (visualize):
+    mat4x4 = visualize_actors( im2.GetOutputPort(), im1.GetOutputPort() )
+  else:
+    mat4x4 = vtk.vtkMatrix4x4()
+
+  transform = vtk.vtkTransform()
+  transform.SetMatrix(mat4x4)
+  
+  transformFilter = vtk.vtkTransformFilter()
+  transformFilter.SetInputConnection( im2.GetOutputPort() )
+  transformFilter.SetTransform( transform )
+  transformFilter.Update()
+  
+  final_image = joinPolyData( transformFilter.GetOutput(), im1.GetOutput() )
+  
+  write_stl( final_image.GetOutputPort(), output_file, vtk.vtkMatrix4x4() )
+  
+def boolean_stl(input_file1, input_file2, output_file, transform_file, operation, visualize, overwrite, func):
+
+  if os.path.isfile(output_file) and not overwrite:
+    result = input('File \"{}\" already exists. Overwrite? [y/n]: '.format(output_file))
+    if result.lower() not in ['y', 'yes']:
+      print('Not overwriting. Exiting...')
+      os.sys.exit()
+  
+  im1 = vtk.vtkSTLReader()
+  im1.SetFileName(input_file1)
+  im1.Update()
+  im1 = cleanPolyData( im1.GetOutput() )
+  
+  im2 = vtk.vtkSTLReader()
+  im2.SetFileName(input_file2)
+  im2.Update()
+  im2 = cleanPolyData( im2.GetOutput() )
+  
+  im2 = applyTransform(transform_file, im2)
+  
   if (visualize):
+    mat4x4 = visualize_actors( im2.GetOutputPort(), im1.GetOutputPort() )
+  else:
+    mat4x4 = vtk.vtkMatrix4x4()
+
+  transform = vtk.vtkTransform()
+  transform.SetMatrix(mat4x4)
+  
+  transformFilter = vtk.vtkTransformFilter()
+  transformFilter.SetInputConnection( im2.GetOutputPort() )
+  transformFilter.SetTransform( transform )
+  transformFilter.Update()
+
+  booleanOperation = vtk.vtkBooleanOperationPolyDataFilter()
+  booleanOperation.SetInputData( 0, im1.GetOutput() )
+  booleanOperation.SetInputData( 1, transformFilter.GetOutput() )
+  if "union" in operation:
+    booleanOperation.SetOperationToUnion()
+  elif "intersection" in operation:
+    booleanOperation.SetOperationToIntersection()
+  elif "difference" in operation:
+    booleanOperation.SetOperationToDifference()
+  else:
+    message("ERROR: Invalid boolean operation.")
+    exit()
```

Comment:
Again, probably don't want to exit here. Instead, throw error or return.

```suggestion
    raise ValueError('Invalid boolean operation: ' + operation)
```